packages/cornucopia/src/stores/localstorage.ts

```diff
@@ -26,21 +21,30 @@ export class CashuLocalStorage implements CashuStorage<StartProofSelection> {
     localStorage.setItem(this.PROOFS_KEY, JSON.stringify(proofs));
   }
 
+  private async selectProofs(filter: ProofFilter) {
+    let proofs = this.loadAllProofs();
+
+    const unlocked = await this.locker.getUnlocked(proofs);
+    if (filter.locked) proofs = proofs.filter((p) => !unlocked.includes(p));
+    else proofs = unlocked;
+
+    if (filter.expired) proofs = await this.locker.getExpired(proofs);
+
+    if (filter.amount !== undefined) {
+      const { send } = selectProofsToSend(proofs, filter.amount);
+      proofs = send;
+      if (sumProofs(proofs) < filter.amount) throw new Error("Failed to find enough proofs");
```

Comment:
Note: there's an InsufficientFundsError in packages/cornucopia/src/errors.ts

I think its helpful to have this specific error so that clients can more easily handle different failure scenarios